core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -475,6 +475,24 @@ else if(segment.startsWith("{") && segment.endsWith("}")) {
 						messageId = messageIdHeader;
 					}
 				}
+				if(StringUtils.isNotEmpty(listener.getHeaderParams())) {
+					String params[] = listener.getHeaderParams().split(",");
+					for (String headerParam : params) {
+						String param = request.getHeader(headerParam);
+						if(param != null) {
+							messageContext.put(headerParam, param);
+						}
+					}
+				}
+				if(StringUtils.isNotEmpty(listener.getCookieParams())) {
+					String params[] = listener.getCookieParams().split(",");
+					for (String cookieParam : params) {
+						Cookie cookie = CookieUtil.getCookie(request, cookieParam);
+						if(cookie != null) {
+							messageContext.put(cookieParam, cookie.getValue());
+						}
+					}
+				}
```

Comment:
We don't do anything with cookies, if you are going to implement this, at least put a restriction on it. Don't just pass all cookies but only those set in the listener.

Reply:
I agree with Niels. I think it is better we comment this code out for now.